scripts/format_notebooks.py

```diff
@@ -181,6 +181,58 @@ def preprocess(self, nb, resources):
         return nb, resources
 
 
+class VersionCheckPreprocessor(preprocessors.Preprocessor):
+    metadata_tag = "VersionCheck"  # tag for added cells so that we can find them easily
+
+    try:
+        # determine the current stellargraph version by importing the library's version file
+        sys.path.append(
+            (
+                os.path.abspath(
+                    os.path.join(
+                        os.path.dirname(__file__), os.path.pardir, "stellargraph"
+                    )
+                )
+            )
+        )
+        import version as sgversion
+
+        stellargraph_version = sgversion.__version__
+    except ImportError as error:
+        print(f"ERROR: unable to import StellarGraph version: {error.message}")
+        stellargraph_version = None
+
+    version_check_code = f"""\
```

Comment:
This might be a case where it's clearer to use `%`, because of the nested `f` string, e.g.:

```python
version_check_code = """\
...
if version.parse(sg.__version__) < version.parse("%(stellargraph_version)s"):
    raise ValueError(
         f"The installed version {sg.__version__} of StellarGraph is older than the version %(stellargraph_version)s required for this notebook. Please update StellarGraph, eg: pip install --upgrade stellargraph"
    )""" % {"stellargraph_version": stellargraph_version}
```

Totally up to you whether you go for this, though.

Reply:
I think both are confusing! 🤷‍♂️